device/samples/pnp/temperature_controller.js

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+const Mqtt = require('azure-iot-device-mqtt').Mqtt;
+const ProvisioningMqtt = require('azure-iot-provisioning-device-mqtt').Mqtt;
+const Client = require('azure-iot-device').Client;
+const ClientPropertyCollection = require('azure-iot-device').ClientPropertyCollection;
+const generateWritablePropertyResponse = require('azure-iot-device').generateWritablePropertyResponse;
+const SymmetricKeySecurityClient = require('azure-iot-security-symmetric-key').SymmetricKeySecurityClient;
+const ProvisioningDeviceClient = require('azure-iot-provisioning-device').ProvisioningDeviceClient;
+
+// Must be "DPS" or "connectionString"
+const deviceSecurityType = process.env.IOTHUB_DEVICE_SECURITY_TYPE;
+
+// Must be set if IOTHUB_DEVICE_SECURITY_TYPE environment variable is "connectionString"
+let deviceConnectionString = process.env.IOTHUB_DEVICE_CONNECTION_STRING;
+
+// Must be set if IOTHUB_DEVICE_SECURITY_TYPE environment variable is "DPS"
+const provisioningHost = process.env.IOTHUB_DEVICE_DPS_ENDPOINT || 'global.azure-devices-provisioning.net';
+const idScope = process.env.IOTHUB_DEVICE_DPS_ID_SCOPE;
+const registrationId = process.env.IOTHUB_DEVICE_DPS_REGISTRATION_ID;
+const symmetricKey = process.env.IOTHUB_DEVICE_DPS_REGISTRATION_KEY;
```

Comment:
Before, these environment variables were `IOTHUB_DEVICE_DPS_DEVICE_ID` and `IOTHUB_DEVICE_DPS_DEVICE_KEY`. This naming seems incorrect to me, so I changed it. However, I think the other languages use the old name. Not sure what we should be using here.

Reply:
As I recall, program managers spent a lot of time working on these names.  Do you REALLY want to be in disagreement with the other SDKs on this?

Reply:
Changed it back.